src/integration/tests/reshard.rs

```diff
@@ -0,0 +1,312 @@
+	where
+		F: Fn(TestArgs) -> T,
+		T: std::future::Future<Output = ()>,
+	{
+		let test_id = format!("{:016x}", thread_rng().gen::<u64>());
+		let mut process_handles: Vec<ChildWrapper> = vec![];
+		let mut file_handles: Vec<PathBuf> = vec![];
+		let mut test_args = TestArgs::default();
+
+		if self.setup_reshard {
+			// Socket paths
+			let app_sock =
+				PathBuf::from(format!("./{test_id}.reshard.app.sock"));
+			let enc_sock =
+				PathBuf::from(format!("./{test_id}.reshard.enclave.sock"));
+			file_handles.extend([app_sock.clone(), enc_sock.clone()]);
+
+			// Minimal manifest envelope
+			let manifest_path =
+				PathBuf::from(format!("./{test_id}.manifest_envelope"));
+			write_minimal_manifest(&manifest_path);
+			file_handles.push(manifest_path.clone());
+
+			// 1) simulator_enclave
+			let sim: ChildWrapper = Command::new(SIMULATOR_ENCLAVE_PATH)
+				.arg(&enc_sock)
+				.arg(&app_sock)
+				.spawn()
+				.expect("spawn simulator_enclave")
+				.into();
+			process_handles.push(sim);
+
+			// 2) reshard_app
+			let quorum_secret = "./fixtures/reshard/quorum.secret";
+			let ephemeral_secret = "./fixtures/reshard/ephemeral.secret";
+			let new_share_set_json = std::fs::read_to_string(
+				"./fixtures/reshard/new-share-set/new-share-set.json",
+			)
+			.expect("read new-share-set.json");
+			let app: ChildWrapper = Command::new("../target/debug/reshard_app")
+				.arg("--usock")
+				.arg(&app_sock)
+				.arg("--quorum-file")
+				.arg(quorum_secret)
+				.arg("--ephemeral-file")
+				.arg(ephemeral_secret)
+				.arg("--manifest-file")
+				.arg(&manifest_path)
+				.arg("--new-share-set")
+				.arg(&new_share_set_json)
+				.arg("--mock-nsm")
+				.spawn()
+				.expect("spawn reshard_app")
+				.into();
+			process_handles.push(app);
+
+			// 3) reshard_host
+			let host_port =
+				qos_test_primitives::find_free_port().expect("find free port");
+			let host: ChildWrapper =
+				Command::new("../target/debug/reshard_host")
+					.arg("--host-ip")
+					.arg(LOCAL_HOST)
+					.arg("--host-port")
+					.arg(host_port.to_string())
+					.arg("--usock")
+					.arg(&enc_sock)
+					.spawn()
+					.expect("spawn reshard_host")
+					.into();
+			process_handles.push(host);
+			qos_test_primitives::wait_until_port_is_bound(host_port);
+
+			let host_addr = format!("http://{LOCAL_HOST}:{host_port}");
+			test_args.reshard_client_addr = Some(host_addr.clone());
+
+			let reshard = ReshardServiceClient::connect(host_addr)
+				.await
+				.unwrap()
+				.max_decoding_message_size(GRPC_MAX_RECV_MSG_SIZE);
+			test_args.reshard_client = Some(reshard);
+		}
+
+		// Run the user test and ensure cleanup.
+		let res =
+			std::panic::AssertUnwindSafe(test(test_args)).catch_unwind().await;
+
+		for p in file_handles {
+			let _ = fs::remove_file(p);
+		}
+
+		assert!(res.is_ok(), "test body panicked");
+	}
+}
+
+fn write_minimal_manifest(path: &PathBuf) {
+	let env = ManifestEnvelope {
+		manifest: Manifest { ..Default::default() },
```

Comment:
You could add a default impl to the `Manifest` so this just can be `let env = ManifestEnvelope::default()`

Reply:
And if easy gate the default impl to some test only feature